src/eureka/S1_detector_processing/s1_process.py

```diff
@@ -97,6 +106,17 @@ def rampfitJWST(eventlabel, ecf_path=None):
     total = (time_pkg.time() - t0) / 60.
     log.writelog('\nTotal time (min): ' + str(np.round(total, 2)))
 
+    # Store citations to relevant dependencies in the meta file
+    # get currently imported modules (top level only)
+    mods = np.unique([mod.split('.')[0] for mod in sys.modules.keys()])
+    
+    # get modules for which we have citations
+    citemods = np.intersect1d(mods, list(CITATIONS))
+
+    # concatenate the bibitems into a bibliography
+    meta.citations = np.append(citemods, meta.inst)
+    meta.bibliography = np.concatenate([CITATIONS[entry] for entry in citemods])
+
```

Comment:
Rather than copy-pasting this code in several places, can you write a function to do most/all of this?

Reply:
These issues should be fixed now.